GetOnlineMerchants/optional_product_category_list_middleware.ts

```diff
@@ -56,30 +58,28 @@ const CommaSeparatedListOf = (
     String
   );
 
-export const OptionalProductCategoryListMiddleware = (
-  name: string
-): IRequestMiddleware<
-  "IResponseErrorValidation",
-  Option<ReadonlyArray<ProductCategory>>
-> => async (
-  request
+export const OptionalProductCategoryListMiddleware = (name: string) => async (
+  request: Request
 ): Promise<
-  Either<
+  E.Either<
     IResponse<"IResponseErrorValidation">,
-    Option<ReadonlyArray<ProductCategory>>
+    O.Option<ReadonlyArray<ProductCategory>>
   >
 > =>
-  taskEither
-    .of<IResponse<"IResponseErrorValidation">, Option<unknown>>(
-      fromNullable(request.query[name])
-    )
-    .chain(maybeQuery =>
-      maybeQuery.foldL(
-        () => taskEither.of(none),
-        query =>
-          fromEither(
-            CommaSeparatedListOf(productCategoryCodec).decode(query)
-          ).bimap(ResponseErrorFromValidationErrors(ProductCategory), some)
+  pipe(
+    TE.of<IResponseErrorValidation, O.Option<unknown>>(
```

Comment:
Do we need this?

Reply:
fixed in https://github.com/pagopa/io-functions-cgn-operator-search/pull/28/commits/8386c96583ac277673207742ac223f1fe84047fc